nibabel/tests/test_loadsave.py

```diff
@@ -58,6 +58,14 @@ def test_file_not_found():
     assert_raises(FileNotFoundError, load, 'does_not_exist.nii.gz')
 
 
+def test_load_empty_image():
+    with InTemporaryDirectory():
+        open('empty.nii', 'w').close()
+        with assert_raises(ImageFileError) as err:
+            load('empty.nii')
```

Comment:
I'm surprised this works on Python 3 - I thought that the `err` variable went out of scope.

Reply:
It seems to work - didn’t dig into why

Reply:
From the docs:

```
Signature: unittest.TestCase.assertRaises(self, expected_exception, *args, **kwargs)
Docstring:
Fail unless an exception of class expected_exception is raised
by the callable when invoked with specified positional and
keyword arguments. If a different type of exception is
raised, it will not be caught, and the test case will be
deemed to have suffered an error, exactly as for an
unexpected exception.

If called with the callable and arguments omitted, will return a
context object used like this::

     with self.assertRaises(SomeException):
         do_something()

An optional keyword argument 'msg' can be provided when assertRaises
is used as a context object.

The context manager keeps a reference to the exception as
the 'exception' attribute. This allows you to inspect the
exception after the assertion::

    with self.assertRaises(SomeException) as cm:
        do_something()
    the_exception = cm.exception
    self.assertEqual(the_exception.error_code, 3)
File:      /anaconda3/lib/python3.6/unittest/case.py
Type:      function
```